src/run.ts

```diff
@@ -19,15 +19,46 @@ type Artifact = InputArtifact & {
 export default async function run(): Promise<void> {
   try {
     const inputs = getInputs()
-    const artifactsToUpload: Artifact[] = inputs.artifacts.map(artifact => ({
+    const artifacts: Artifact[] = inputs.artifacts.map(artifact => ({
       ...artifact,
       mime_type: mimeTypeFromExtension(extname(artifact.path).toLowerCase()),
       external_id: uuidv4()
     }))
+
+    const [artifactsWithFiles, artifactsWithoutFiles] = artifacts.reduce<
+      [Artifact[], Artifact[]]
+    >(
+      ([withFiles, withoutFiles], artifact) => {
+        if (existsSync(artifact.path)) {
+          return [[...withFiles, artifact], withoutFiles]
+        } else {
+          return [withFiles, [...withoutFiles, artifact]]
+        }
+      },
+      [[], []]
+    )
+
+    if (inputs.ifFilesNotFound === 'warn') {
+      for (const artifact of artifactsWithoutFiles) {
+        core.warning(
+          `Artifact file not found at '${artifact.path}' for artifact '${artifact.name}'`
+        )
+      }
+    } else if (inputs.ifFilesNotFound === 'error') {
+      for (const artifact of artifactsWithoutFiles) {
+        core.error(
+          `Artifact file not found at '${artifact.path}' for artifact '${artifact.name}'`
+        )
+      }
+      if (artifactsWithoutFiles.length) {
+        core.setFailed('Artifact(s) are missing file(s)')
+      }
+    }
+
     const bulkArtifactsResult = await createBulkArtifacts(
       {
         account_name: inputs.accountName,
-        artifacts: artifactsToUpload.map(artifact => ({
+        artifacts: artifacts.map(artifact => ({
           kind: artifact.kind,
```

Comment:
I made the decision to still tell Vanguard about the artifacts we're _trying_ to upload. In the end, they'll be marked as abandoned, but at least we'll be able to see how often it's happening. Happy to swap this up, though!